src/coord/rmr/command.c

```diff
@@ -218,3 +224,34 @@ void MRCommand_ReplaceArgSubstring(MRCommand *cmd, int index, size_t pos, size_t
 void MRCommand_SetProtocol(MRCommand *cmd, RedisModuleCtx *ctx) {
   cmd->protocol = is_resp3(ctx) ? 3 : 2;
 }
+
+void MRCommand_PrepareForSlotInfo(MRCommand *cmd, uint32_t pos) {
```

Comment:
I understand u need this preparation to make sure that the copy handles it, and then in uvthread we can subsititute for the specific shard's info?

Reply:
Exactly. And also `FT.SEARCH` reuses the same command for all the shards so it calls `MRCommand_PrepareForSlotInfo` once and `MRCommand_SetSlotInfo` for each shard

Reply:
I will improve the comments here